frontend/app/src/components/Header/Header.tsx

```diff
@@ -16,33 +16,27 @@
 
 import React, { ReactElement, ReactNode } from "react"
 
-import { useAppContext } from "@streamlit/app/src/components/StreamlitContextProvider"
-
 import {
   StyledHeader,
   StyledHeaderDecoration,
   StyledHeaderToolbar,
 } from "./styled-components"
 
 export interface HeaderProps {
+  showToolbar: boolean
+  showColoredLine: boolean
   children: ReactNode
-  isStale?: boolean
 }
 
-function Header({ isStale, children }: Readonly<HeaderProps>): ReactElement {
-  const { wideMode, embedded, showToolbar, showColoredLine } = useAppContext()
-
-  let showHeader = true
-  if (embedded) {
-    showHeader = showToolbar || showColoredLine
-  }
```

Comment:
This section used to control setting `display` to `block` or `none` - Moved logic to `App` to control via conditional rendering instead